mhlib/app.go

```diff
@@ -269,3 +293,81 @@ func (a *App) render(configFile string) (*string, *string, *[]byte, error) {
 
 	return &chart, &chartVersion, &overrides, nil
 }
+
+
+func selfRender(templateValuesStr string, enableGomplate bool) (string, error) {
+	/*
+	This function will accept an input string and run it through the
+	  templating engine as both the values dictionary as well as the
+	  template string.  It will repeat the process until the templating
+	  result stops changing.
+	*/
+
+	type Values map[string]interface{}
+
+	lastRender := templateValuesStr
+	for i := 0; i < 10; i++ {
+
+		// Unmarshal the file as a values dict
+		vals := Values{}
+		err := yaml.Unmarshal([]byte(templateValuesStr), &vals)
+		if err != nil { panic(err) }
+
+		tmpl := template.New("SelfTemplate")
+		tmpl.Delims("[[", "]]")
+
+		if enableGomplate {
+			// Read the defined datasources and datasourcehaders
+			//   from the values dict
+			dataSources := []string{}
+			dataSourceHeaders := []string{}
+			if _, ok := vals["gomplate"]; ok {
+				tmp := vals["gomplate"].(map[string]interface{})
+				if _, ok := tmp["datasources"]; ok {
+					dataSources = convertInterfaceListToStringList(
+						tmp["datasources"].([]interface{}))
```

Comment:
I'm a little rusty on my Golang, but it seems we're not handling the possibility of error during type conversion.  Agreed?

Reply:
converInterfaceListToStringList uses `fmt.sprint` which will always succeed in printing the string representation of any object.

However, I've fixed error handling everywhere else, esp in the self-render function

Reply:
Does that really work?  Right now the "ok" is checking the existence of the key "datasources" inside of the map "tmp".  Once you do the type assert, I think the "ok" is checking the type assertion and not the existence of the key in the map.  Can you confirm @drekle